xarray/core/utils.py

```diff
@@ -715,28 +715,32 @@ def __len__(self) -> int:
         return len(self._data) - num_hidden
 
 
-def infix_dims(dims_supplied: Collection, dims_all: Collection) -> Iterator:
+def infix_dims(
+    dims_supplied: Collection, dims_all: Collection, missing_dims: str = "raise"
+) -> Iterator:
     """
-    Resolves a supplied list containing an ellispsis representing other items, to
+    Resolves a supplied list containing an ellipsis representing other items, to
     a generator with the 'realized' list of all items
     """
     if ... in dims_supplied:
         if len(set(dims_all)) != len(dims_all):
             raise ValueError("Cannot use ellipsis with repeated dims")
-        if len([d for d in dims_supplied if d == ...]) > 1:
+        if sum(d is ... for d in dims_supplied) > 1:
```

Comment:
Are we sure it should be `is` rather than `==`? I somewhat recall having this discussion before, so would want to check.

Reply:
@max-sixty It was `==` and I change, for some reason the IDE was complaining about it. I'm gonna change it back

Reply:
`...` is a singleton object (https://www.geeksforgeeks.org/what-is-three-dots-or-ellipsis-in-python3/), so like when comparing to `None` I think the 'right' thing to do is `is ...`.

Reply:
I can't find the old discussion so maybe I'm imagining it.

@keewis might I remember that you commented on this before?

Otherwise `is` indeed seems more likely to be correct

Reply:
the only discussion I can remember is #3900, and that was about strings and formatting (did we finish that, by the way?).

I agree that `x is ...` (or `x is Ellipsis`) is the correct way to compare.

Reply:
however, using `sum` to count occurences of `...` seems a bit strange. Maybe use `list.count` instead?

Reply:
@keewis I think that dims_supplied is supposed to be a Collection not a list, as far as I know, there is no method count on Collection but I could be mistaken. The other way to check, is like it was previously, create a list and get the length both ways are common approaches for checking the length of iterables, this just the way I preferred (there is no need to create a list just to compute the length) but I'm happy to change it back if you deem necessary

Reply:
well, we cast to `set` anyways, so I guess casting to `list` is not an issue. Also, I think
```python
list(dims_supplied).count(...)
```
is easier to understand than both the current
```python
len([x for x in dims_supplied if x is ...])
```
and
```python
sum([x is ... for x in dims_supplied])
```
but none of these are particularly difficult to read, so I guess it's up to you.